src/main/java/client/Character.java

```diff
@@ -3107,11 +3149,29 @@ private synchronized void gainExpInternal(long gain, int equip, int party, boole
             if (leftover > 0) {
                 gainExpInternal(leftover, equip, party, false, inChat, white);
             } else {
+                saveExpLogToDB();
                 lastExpGainTime = System.currentTimeMillis();
             }
         }
     }
 
+    private void saveExpLogToDB() {
+        if (YamlConfig.config.server.USE_EXP_GAIN_LOG) {
+                Map<String, Object> psMap = new HashMap<>();
```

Comment:
Should create a new class to store this information in a type safe manner. I suggest a record class.